### PR TITLE
feat(dato-select): add group filtering for dato-select component

### PR DESCRIPTION
**Pull request by orgilor**

### CODE REVIEW

File: lib/src/select/search.strategy.ts [hunk 1]
```diff
@@ -5,22 +5,23 @@
...
  * found in the LICENSE file at https://github.com/datorama/client-core/blob/master/LICENSE
  */
-import { DatoOptionComponent } from '../options/option.component';
 
 /**
  * DatoSelectSearchStrategy - Interface for custom search strategies
  */
 export interface DatoSelectSearchStrategy {
-  (datoOption: DatoOptionComponent, searchValue: string, labelKey?: string): boolean;
+  (searchIn: any, searchValue: string, labelKey?: string): boolean;
```
Comment (NetanelBasal) [inline lib/src/select/search.strategy.ts]: why any?

File: lib/src/select/search.strategy.ts [hunk 2]
```diff
@@ -5,22 +5,23 @@
...
  * The default search strategy is compare lowercased strings
- * @param {DatoOptionComponent} datoOption
+ * @param {any} searchIn
  * @param {string} searchValue
  * @param {string} labelKey
  * @returns {boolean}
  */
-export const defaultClientSearchStrategy: DatoSelectSearchStrategy = (datoOption: DatoOptionComponent, searchValue: string, labelKey: string) => {
-  return datoOption.option[labelKey].toLowerCase().indexOf(searchValue.toLowerCase()) > -1;
+export const defaultClientSearchStrategy: DatoSelectSearchStrategy = (searchIn: any, searchValue: string, labelKey: string) => {
```
Comment (NetanelBasal) [inline lib/src/select/search.strategy.ts]: why any?

File: lib/src/select/select.component.ts [hunk 3]
```diff
@@ -468,6 +481,27 @@ export class DatoSelectComponent extends BaseCustomControl implements OnInit, On
    * Show all options
    */
   private showAll() {
+    const isAccordion = this.isAccordionGroup();
+    const groupComponents: ListGroupComponent[] = this.getGroupComponentsArray();
+
+    groupComponents.forEach((groupComponent: ListGroupComponent) => {
```
Comment (NetanelBasal) [inline lib/src/select/select.component.ts]: use for of

File: lib/src/select/select.component.ts [hunk 4]
```diff
@@ -468,6 +481,27 @@ export class DatoSelectComponent extends BaseCustomControl implements OnInit, On
...
+      groupComponent._hidden = false;
+
+      /* If this accordion group has been expended by the search function,
+         then collapse it to the original state
+      */
+      if (isAccordion) {
+        const anyGroup = groupComponent as any;
```
Comment (NetanelBasal) [inline lib/src/select/select.component.ts]: why any

File: lib/src/select/select.component.ts [hunk 5]
```diff
@@ -480,19 +514,102 @@ export class DatoSelectComponent extends BaseCustomControl implements OnInit, On
...
+   * Search for matching groups and children.
+   * @param {ListGroupComponent[]} groupComponentsArray
+   * @param {string} value
+   * @return {boolean}
+   */
+  private searchInGroups(groupComponentsArray: ListGroupComponent[], value: string): boolean {
+    const isAccordion = this.isAccordionGroup();
+    let hasResults = false;
+
+    this._data.forEach((group, index) => {
```
Comment (NetanelBasal) [inline lib/src/select/select.component.ts]: use normal for (faster)

File: lib/src/select/search.strategy.ts (hunk 1 above)
Comment (orgilor) [reply lib/src/select/search.strategy.ts]: thats what i copied form the dato list

File: lib/src/select/search.strategy.ts (hunk 2 above)
Comment (orgilor) [reply lib/src/select/search.strategy.ts]: same

File: lib/src/select/select.component.ts (hunk 4 above)
Comment (orgilor) [reply lib/src/select/select.component.ts]: same

File: lib/src/select/select.component.ts (hunk 3 above)
Comment (orgilor) [reply lib/src/select/select.component.ts]: same

**Pull request merged**